index.js

```diff
@@ -171,9 +188,22 @@ function configure (options) {
   }
   const circularValue = getCircularValueOption(options)
   const bigint = getBooleanOption(options, 'bigint')
-  const deterministic = getBooleanOption(options, 'deterministic')
+  const deterministicOption = getDeterministicOption(options, 'deterministic')
+  const deterministic = typeof deterministicOption === "function" ? true : deterministicOption
```

Comment:
```suggestion
  const deterministic = getDeterministicOption(options, 'deterministic')
  const comparator = typeof deterministic === "function" ? deterministic : undefined
```

In case deterministic is a function `if (deterministic)` is going to be truthy and things continue to work as they would otherwise.